pkg/printers/humanreadable_flags.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package printers
+
+import (
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/kubectl/scheme"
+)
+
+// HumanPrintFlags provides default flags necessary for template printing.
+// Given the following flag values, a printer can be requested that knows
+// how to handle printing based on these values.
+type HumanPrintFlags struct {
+	NoHeaders *bool
+
+	// TODO: this is deprecated - remove this field in next version
+	ShowAll *bool
+
+	ShowKind     *bool
+	ShowLabels   *bool
+	ColumnLabels *[]string
+	SortBy       *string
+
+	AbsoluteTimestamps bool
+	WithNamespace      bool
+	Kind               string
+}
+
+// ToPrinter receives an outputFormat and returns a printer capable of
+// handling human-readable output.
+func (f *HumanPrintFlags) ToPrinter(outputFormat string) (ResourcePrinter, bool, error) {
+	if len(outputFormat) > 0 && outputFormat != "wide" {
+		return nil, false, nil
+	}
+
+	encoder := scheme.Codecs.LegacyCodec(scheme.Registry.EnabledVersions()...)
+	decoder := scheme.Codecs.UniversalDecoder()
+
+	p := NewHumanReadablePrinter(encoder, decoder, PrintOptions{})
+
+	if f.NoHeaders != nil {
+		noHeaders := *f.NoHeaders
+		if !noHeaders {
+			p.EnsurePrintHeaders()
+		}
+	}
+
+	// TODO: handle sorting. importing "kubectl.SortingPrinter"
+
+	return p, true, nil
+}
+
+// AddFlags receives a *cobra.Command reference and binds
+// flags related to human-readable printing to it
+func (f *HumanPrintFlags) AddFlags(c *cobra.Command) {
+	if f.NoHeaders != nil {
+		c.Flags().BoolVar(f.NoHeaders, "no-headers", *f.NoHeaders, "When using the default or custom-column output format, don't print headers (default print headers).")
```

Comment:
@soltysh @deads2k The `no-headers` flag is also bound by custom-columns PrintFlags here: https://github.com/kubernetes/kubernetes/pull/61501/files#diff-031dfa2edd2855fceafd54e790b79f1cR102

Reply:
Yeah, I guess we need to figure out the best way to solve this problem.

Reply:
Went ahead and added a parameter that sets this on the constructor func for the PrintFlags struct